nodes/src/nodes/embedding_native/native_embedding.cpp

```
@@ -0,0 +1,393 @@
+extern "C" {
+
+/**
+ * Initialize the embedding engine.
+ * model_path: path to .onnx model file
+ * vocab_path: path to vocab.txt file
+ * embedding_dim: output embedding dimension (e.g. 384 for MiniLM)
+ * max_length: max token sequence length
+ * num_threads: intra-op parallelism (0 = default)
+ * Returns: 0 on success, -1 on error.
+ */
+int32_t embedding_init(
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)